src/js/containers/AppList.js

```diff
@@ -30,25 +30,39 @@ const mapStateToProps = (state) => {
         var name = app.isCloudApplication ? app.appName + " (Cloud)" : app.appName;
         var devicename = (app.deviceInfo.name.trim()) ? app.deviceInfo.name 
             : app.deviceInfo.transportType + ": " + app.deviceInfo.id.substring(0, 10) + "...";
+
+        var greyOut = app.greyOut;
+        if (app.appID === state.pendingAppLaunch) {
+            name = 'Loading...';
+            greyOut = true;
+        }
+
         return {
             appID: app.appID,
             class: 'with-image',
             name: name,
             devicename: devicename,
             image: icon,
-            link: '/' + link,
+            link: undefined,
             cmdID: app,
-            greyOut: app.greyOut
+            greyOut: greyOut
         }
     })
 
-    return {data: data}
+    return {data: data, pendingAppLaunch: state.pendingAppLaunch}
```

Comment:
```suggestion
    return {data: data}
```

There is no reason for HScrollMenu to receive this prop.